timesketch/api/v1/resources/upload.py

```diff
@@ -62,74 +126,55 @@ def _upload_and_index(
             A timeline if created otherwise a search index in JSON (instance
             of flask.wrappers.Response)
         """
-        # Check if search index already exists.
-        searchindex = SearchIndex.query.filter_by(
+        searchindex = self._get_index(
+            name=timeline_name,
+            description=timeline_name,
+            sketch=sketch,
+            index_name=index_name,
+            data_label=data_label,
+            extension=file_extension)
+        searchindex.set_status('processing')
+
+        timeline = Timeline.get_or_create(
             name=timeline_name,
+            description=timeline_name,
+            sketch=sketch,
             user=current_user,
-            index_name=index_name).first()
+            searchindex=searchindex)
```

Comment:
This means that if the timeline description has changed, a new timeline will be created. And also, if I try to append to an existing timeline that someone else created then a new timeline with the same name will be created but with the current user as creator.

Should we do something like this instead?

```
timeline = Timeline.query.filter_by(name=timeline_name,sketch=sketch,searchindex=searchindex).first()
```

WDYT?

Reply:
better?